### ISSUE TITLE
[docs]: Add a style guide for documentation

### ISSUE THREAD

**Issue by BenJamesBen** [freesewing/freesewing#3389]
### Are you a FreeSewing patron? 😃

No, I am not 😞

### Suggested content

Add a style guide for FreeSewing documentation to help doc writers maintain consistency.

**Comment by BenJamesBen** [comment]
A draft style guide that I have been using is below. (I am still unsure about some of the preferred spellings.)

- Regarding FreeSewing:
  - The spelling for the name of the organization is "FreeSewing", not "FreeSewing.org", and the "F" and "S" are capitalized.
  - The websites are "freesewing.org" and "freesewing.dev", with all letters in lowercase.
  - "freesewing" without capitalization is used for package names and the GitHub account name.

- Preferred spellings that might disagree with dictionary spellings:
  - "lifecycle" when specifically referring to the stages that the FreeSewing technology goes through when it runs and generates a pattern.
  - "whitespace" when specifically used to describe computer whitespace characters, (while "white space" is used to describe spacing before/after words or lines).
  - "bartack" when specfically referring to the element from the FreeSewing plugin, ("bar tack" when otherwise referring to the element) (Or, is the spelling "bartack" accepted sewing jargon?)
  - "cutonfold" when specifically referring to the element from the FreeSewing plugin, ("cut-on-fold" when refering to the line in general)

- Jargon that might not be found in dictionaries but are accepted words/spelling:
  - Sewing and fashion: "womenswear", "menswear", "sleevecap", "grainline", "outseam", "underbust"
  - Technology: "truthy", "falsy", "backend", "frontend", "config", "monorepo", "serverless", "runtime", "lockfile"
  - FreeSewing-specific variations: "Howto" ("H" capitalized), "measies", "scalebox", "miniscale", "bnotch", "cutlist", "multiset"
  - Other acceptable words: - "startpoint", "endpoint" (Or, should we standardize on "start point" and "end point"?)

- Other preferred spellings:
  - "stand-alone", not "standalone".
  - "line break", not "linebreak".
  - "website", not "web site".
  - "pop-up", not "popup"

- There should be a space character between numbers and SI units. For example, "5 cm" and "1.3 mm", rather than "5cm" and "1.3mm". (The exception is for CSS and SVG values. Their specifications say that the space character should not be used.)

- Correct capitalizations for proper nouns and abbreviations:
  - SVG, JSON, YAML (all letters capitalized)
  - NextJS, ReactJS (the first letter and "JS" capitalized for these)
  - Frontmatter, Strapi, Crowdin, Prisma, Markdown, Gitpod, Unix, Linux
    (only the first letter capitalized for these)
  - SQLite (the "SQL" capitalized for this)
  - GitHub ("G" and "H" capitalized, not "Github")
  - JavaScript ("J" and "S" capitalized, not "Javascript" or "javascript")
  - macOS ("OS" capitalized, not "Mac OS" or "mac OS")
  - Node.js ("N" capitalized, "js" lowercase, not "Node" or "Node JS" or "NodeJS")

- Regarding Bézier:
  - "Bézier" with a capital "B" and an acute accent mark over the "é" is used to describe Bézier curves.
  - "Bezier" with a capital "B" and unaccented "e" is used to describe the Bezier class used in code.
  - "bezier" without capitalization is used for package names.

**Comment by SeaZeeZee** [comment]
While I think a style guide or template for documentation would be very useful in lowering the barrier for documentation to be written I don't think our aim should be standardisation across the board. We are an open source project with a large amount of different contributors and I think it is worth preserving peoples personal style of expressing themselves. Spelling varies across different kinds of English and I don't think we gain anything by mandating British English or American English. Not to mention that some if not most of our contributors have English as a second language. 

Additionally I think if we present new contributors with a document on how they *should* spell things that might scare people off. We can work to correct typo's and grammar and spelling mistakes that impede understanding of the text during the process of reviewing pull requests. And if we want to create a style guide I think that is something that the community should have input on. 

That doesn't mean that you can't keep to these spellings personally if you prefer that though! 

I think a style guide that covers things such as layouts and different ways to format documentation would be very useful. I will work on making one for the design documentation so I can share it as a starting point in an upcoming contributor call and we can go from there.

**Comment by BenJamesBen** [comment]
I am torn about this issue. On one hand, I don't want strict rules that make it a hassle to write and update documentation, something that scares people away from participation. On the other hand, having clear guidelines can be very helpful and save people time.

For example, in PR #3390 I just found out that Joost wants both "FreeSewing.org" and "FreeSewing.dev" styled with capital "F" and S". I wish I had had this information beforehand. It would have prevented me from doing it wrong for so long. Instead, I had been changing the "FreeSewing.org" and "FreeSewing.dev" to the incorrect "freesewing.org" and "freesewing.dev" in ever document I updated. Now I have to go through all the documentation again and correct my mistakes. Having the style guide would have prevented this wasted effort.

Updating what I previously posted above, my new personal style guide is now:
- Regarding FreeSewing:
  - "FreeSewing" is styled with the "F" and "S" capitalized.
  - This "FreeSewing" capitalization is also used in website names, for example "FreeSewing.org" and "FreeSewing.dev".
  - The name of the project/group/organization itself is "FreeSewing", not "FreeSewing.org". The term "FreeSewing.org" is used only when referring to the website.
  - However, "freesewing" without capitalization is used for package names, the GitHub account name, and other case-sensitive/coding applications.

**Comment by joostdecock** [comment]
I think it's good to write down the *desired state* of what we prefer to see. That does not mean people are going to be chastised when they write things differently, but it merely means that for those who would like to make the effort to improve consistency, the documentation is there.

So I'm in favor of writing a style guide, because for me it serves the same purpose as our best practices for pattern designers. We advice to do things are certain way, but we don't crucify people when they don't.

The FreeSewing example @BenJamesBen points out is a clear case where such a guide could have prevented wasted work.

So to summarize: I think a document that says: **This is how we prefer you write things** would be a welcome addition.

**Comment by joostdecock** [comment]
Stale. Closing.
